### PR TITLE
Add generics to memoize, partial, and once

### PR DESCRIPTION
**Pull request by cmaddalozzo**

### CODE REVIEW

**Review by charlespwd — APPROVED** [review]
LGTM 👍

File: types/ramda/index.d.ts
Comment (sheetalkamat) [inline types/ramda/index.d.ts]: The order of memoize function overload needs to be reversed so that the default signature is picked up ?

Comment (rbuckton) [reply types/ramda/index.d.ts]: This also could be written as the following, without overloads:

```ts
  memoize<T = any>(fn: (...a: any[]) => T): (...a: any[]) => T;
```

Comment (rbuckton) [inline types/ramda/index.d.ts]: This overload does not have associated documentation. Consider copying the documentation from the preceding overload.

File: types/ramda/index.d.ts
Comment (rbuckton) [inline types/ramda/index.d.ts]: This overload does not have associated documentation. Consider copying the documentation from the preceding overload.

File: types/ramda/index.d.ts
Comment (rbuckton) [inline types/ramda/index.d.ts]: This overload does not have associated documentation. Consider copying the documentation from the preceding overload.

**Review by rbuckton — CHANGES_REQUESTED** [review]
@cmaddalozzo, can you either modify the overload ordering mentioned by @sheetalkamat, or comment as to why this specific ordering was taken? Alternatively, consider the version I mention using type parameter defaults.

**Comment by RyanCavanaugh** [comment]
@cmaddalozzo Please address comments from the code reviewers.

**Comment by cmaddalozzo** [comment]
Hey guys, sorry for the delayed response. I like @rbuckton's suggestion. Updating the PR now.

**Comment by RyanCavanaugh** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Comment by RyanCavanaugh** [comment]
@rbuckton - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Review by rbuckton — APPROVED** [review]

**Pull request merged**
